src/pool/srv_pool.c

```diff
@@ -1812,10 +1788,9 @@ ds_pool_connect_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		D_GOTO(out_map_version, rc);
 
-	if (!permitted(&ugm, in->pci_uid, in->pci_gid, in->pci_capas)) {
-		D_ERROR(DF_UUID": refusing connect attempt for uid %u gid %u "
-			DF_X64"\n", DP_UUID(in->pci_op.pi_uuid), in->pci_uid,
-			in->pci_gid, in->pci_capas);
+	if (!ds_sec_can_pool_connect(&ugm, &in->pci_cred, in->pci_capas)) {
+		D_ERROR(DF_UUID": refusing connect attempt for "
+			DF_X64"\n", DP_UUID(in->pci_op.pi_uuid), in->pci_capas);
```

Comment:
Since ds_sec_can_pool_connect may return an error, 0, or 1, should we log the return value if it's an error?

Reply:
Oh! If it can return an error, this logic is incorrect. -DER_INVAL is just as true as 1 is.

Reply:
Fixed